### ISSUE TITLE
Support for TryRead() methods

### ISSUE THREAD

**Issue by kbilsted** [super-csv/super-csv#23]
I was thinking that maybe we could introduce TryReadxxx() methods on the reader implementations which do not throw exceptions, rather they may return null/an empty list maybe along with a list of strings for the caller to inspect and react upon. In C# "out parameters" are typically used with TryXXX methods. I'm not sure what to use in Java. Especially if we want to keep compatibility with java 1.5

something like

public boolean TryRead(T bean, string[] nameMapping, List<String> rawLineData)
What are your thoughts? If you all think its a nice idea, I would not mind @vleroy to submit a PR for that.

**Comment by ZioberMichal** [comment]
Interesting idea. :+1: 

I think, that if we want to implement something like this we should return much more than only "boolean" or "null/empty" value. User probably would like to see complex `POJO` class with list of columns for which parsing has failed with error messages and all additional data which we can provide.

**Comment by kbilsted** [comment]
Yes that might be a good idea. Originally, when I thought of this I was just thinking about the "standard" try-pattern with a method `object Read() throws ...` and a `bool TryRead(out result)` - but since Java does not have the possibility of returning multiple values natively, we need to wrap the result in some resulttuple or something, so we might at well cater for the need of providing more information now that we've gone through the trouble of trying to read.
